engine/gamesys/src/gamesys/scripts/script_buffer.cpp

```diff
@@ -985,23 +1006,44 @@ namespace dmGameSystem
 
     // copies values from array to (empty) lua table on top of the stack
     template <typename T>
-    static void arrayToLuaTable(lua_State* L, const T* values, uint32_t count, dmBuffer::ValueType valueType) {
+    static void ArrayToLuaTable(lua_State* L, const T* values, uint32_t count, dmBuffer::ValueType valueType) {
         if (valueType == dmBuffer::VALUE_TYPE_FLOAT32) {
-            for (uint32_t i=0; i<count; i++) {
+            for (uint32_t i=0; i<count; i++)
+            {
                 T value = values[i];
                 lua_pushnumber(L, value);
                 lua_rawseti(L, -2, i+1); // lua indices start from 1
             }
-        } else {
-            for (uint32_t i=0; i<count; i++) {
+        } else
+        {
+            for (uint32_t i=0; i<count; i++)
+            {
                 T value = values[i];
                 lua_pushinteger(L, value);
                 lua_rawseti(L, -2, i+1); // lua indices start from 1
             }
         }
     }
 
-    // local metadata, numbertype  = buffer.get_metadata(self.new_buffer, "aabb")
+    /*# retrieve a metadata entry from a buffer
+     *
+     * Get a named metadata entry from a buffer along with its type.
+     *
+     * @name buffer.get_metadata
+     * @param buf [type:buffer] the buffer to get the metadata from
+     * @param metadata_name [type:hash|string] name of the metadata entry
+     * @return values [type:table] table of metadata values or nil if the entry does not exist
+     * @return value_type [type:constant] numeric type of values or nil
```

Comment:
I'm not 100% sure about the syntax here. Perhaps @britzl knows?

Reply:
Yes, using multiple `@return` is correct! I think for type you need to use `number` as that is what will be seen on the Lua side!

Reply:
cool. About the type, `buffer.create` uses [type:constant]. I think it's better to stick to that. Right ?